### DISCUSSION TITLE
Call for testing the new Brands feature

### DISCUSSION THREAD

**Discussion by jimjasson** [woocommerce/woocommerce#51678] [Brands]
In WooCommerce 9.4, we plan to release all the functionality that was previously offered by the [WooCommerce Brands](https://woocommerce.com/products/brands/) plugin as part of WooCommerce core. If you have a moment, we’d love to hear your feedback! 

### What to test

The [nightly build of WooCommerce](https://github.com/woocommerce/woocommerce/releases/tag/nightly) contains the feature. You may upload `woocommerce-trunk-nightly.zip` into a suitable test environment or spin up a [WP Playground](https://bit.ly/woo-coming-soon-mode)!

Since the feature is behind a feature flag for the time being, you may install the following snippet to activate Brands in your store:

```
add_action( 'init', function() {
    update_option( 'wc_feature_woocommerce_brands_enabled', 'yes' );
    update_option( 'woocommerce_remote_variant_assignment', 2 );
} );
```

This can be added directly to the `functions.php` file of your active theme or via a plugin like [Code Snippets](https://href.li/?https://wordpress.org/plugins/code-snippets/).

#### Activation

If you have already been using [WooCommerce Brands](https://woocommerce.com/products/brands/) as a separate extension, then when you upload WooCommerce to the latest `nightly` version and enable Brands functionality in core via the above snippet, you should: 
- see a notice that the Brands plugin has been deactivated:

![image](https://github.com/user-attachments/assets/f4e67144-b9fa-4b6e-baf5-6ba939ad5e8c)

- see a notice that the Brands plugin is no longer needed:

![image](https://github.com/user-attachments/assets/6a87ecea-12d6-4357-afcd-c6087471812e)

- not be able to enable the Brands plugin: 

![image](https://github.com/user-attachments/assets/c26ef736-9ac0-4aab-b2d9-ff339b3a2288)

If you don't have WooCommerce Brands installed as a separate plugin in your store, you shouldn't see any Brands-related notices when updating WooCommerce to the latest `nightly` version. 

#### Creating and managing brands

1. Go to **Products > Brands** and create a new brand. Add a description and an image.
2. Confirm that you can edit the brand you created, change some of its settings and re-save.
3. If you have more than 1 Brands, confirm that you can sort them by dragging and dropping them in the Brands table.
4. Go to the **Products** tab and click to edit a product.
5. Assign a brand to the product.
6. Repeat the above two steps, until you have a couple of products per brand.

#### Brands short-codes

1. If you don’t have 3 Brands in your store already, create at least 3 Brands under **Products > Brands**.
2. Assign some products to each brand.
3. Create a new page under Pages.
4. Give a title to the page.
5. Insert this short-code: `[product_brand_list show_empty_brands="true"]`.
6. View the page and ensure that all the brands in your store show up alphabetically.
7. Edit the page.
8. Insert this short-code: `[brand_products brand="woo" per_page="12" columns="4"]` and replace woo with the slug of one of your brands.
9. View the page and ensure that the products that belong to this brand show up.
10. Edit the page.
11. Insert this short-code: `[product_brand_thumbnails number="12" show_empty="false"]`.
12. View the page and ensure that all brands thumbnails show up.
13. Edit the page.
14. Insert this short-code: `[product_brand_thumbnails_description number="12" show_empty="false"]`.
15. View the page and ensure that all brands thumbnails along with their descriptions show up.
16. Edit a product that belongs to a brand.
17. Insert this short-code: `[product_brand class="alignright"]` in its short description.
18. View the product page and ensure that the brand’s image shows up.

#### Brands widgets

1. Use a theme that supports widgets, such as Storefront.
2. Go to **Appearance > Widgets**.
3. Add the **WooCommerce Brand Description** widget (recommended: to the Sidebar).
4. View a brand’s page.
5. Ensure that the brand’s description shows up where you placed the widget.
6. Add the **WooCommerce Brand Thumbnails** widget (recommended: to the Sidebar).
7. View any page in your store.
8. Ensure that a list of thumbnails of all brands show up where you placed the widget.
9. Add the **WooCommerce Brand Layered Nav** widget (recommended: to the Sidebar).
10. View the Shop page.
11. Ensure that a list of all brands shows up where you placed the widget.
12. Experiment with the widgets’ display settings.

#### Coupons with Brands restrictions

1. Go to **Marketing > Coupons**.
2. Create a new coupon or edit an existing one.
3. Go to the Usage restrictions tab.
4. Locate Product brands and select a brand there, e.g. ‘Woo’.
5. Add a product to the cart that doesn’t belong to the category you selected.
6. Try to apply the coupon and ensure that it doesn’t apply.
7. Add a product to the cart that belongs to the category you selected.
8. Try to apply the coupon and ensure that it applies.

#### Brands Blocks

1. Use a theme that supports blocks, for example Twenty Twenty Four.
2. Go to **Appearance > Editor**.
3. Edit the Single Product template.
4. Add the Brands block.
5. View a product that belongs to a brand.
6. Ensure that the brand’s name shows up where you placed your block.

#### REST API

1. Go through all these examples: [https://woocommerce.com/document/woocommerce-brands/#rest-api-support](https://href.li/?https://woocommerce.com/document/woocommerce-brands/#rest-api-support) and ensure that Brands are created, deleted, retrieved as expected.

### Known issues

To ensure a smooth transition of [WooCommerce Brands](https://woocommerce.com/products/brands/) plugin users, we merged the codebase as is, without changing any features or updating any flows. A list of open issues about this feature can be found [here](https://github.com/woocommerce/woocommerce/issues?q=is%3Aopen+is%3Aissue+label%3A%22focus%3A+brands%22).

**Comment by jarman1** [comment]
This is a real positive step in the right direction for Woo. Bringing more esstential features to the core for free. Less plugins to run, more usability. Continued moves like this will keep Woo in the game going forward against players like Shopify and Surecart.

**Reply by jimjasson** [reply]
Thank you for the feedback @jarman1 !

**Comment by itthinx** [comment]
A little Gist plugin added for convenience to enable it https://gist.github.com/itthinx/3907574bede6426d95fa8797a0a4dbbd

**Reply by itthinx** [reply]
... as it just [updates the options](https://developer.woocommerce.com/2024/10/01/introducing-brands/), activating it once is of course enough and can then be deactivated

**Reply by jimjasson** [reply]
Thanks for that @itthinx! 

Heads-up that Brands will be enabled for all users starting with WooCommerce v9.6.

**Comment by betayazilim** [comment]
Hi there,

I use attributes for Brand, EAN and MPN since years.
EAN/GTIN added to core before, now Brand will be added. These are needed features since the beginning.
Thanks

I am sure there are tons of shops who have same usage like me. Is there any way to move EAN and Brand from attribute to woocommerce's area?
It would be great if there is a tool to move data.

Btw, Google merchant center asks 3 things
Brand
EAN (GTIN)
MPN : Manufacturer product number

Since you already handled 2 of them, why not add MPN to core? Just under EAN?

**Reply by jimjasson** [reply]
Hey @ned-bs, 

Glad to hear that you are enjoying the latest additions in WooCommerce core! 🚀

> I am sure there are tons of shops who have same usage like me. Is there any way to move EAN and Brand from attribute to woocommerce's area?

Each store may have a different implementation of brands, either by using one of the numerous brands plugins or by creating a custom system, like you did with attributes. There isn't a migration that would work for all these cases out of the box -- only users who have been previously using brands via the [WooCommerce Brands](https://woocommerce.com/document/woocommerce-brands/) extension will see their brands automatically migrated. The reason why is that the same `product_brand` taxonomy is used. If you'd like to create a custom migration process for your setup, then I recommend implementing a method to move brands information from your custom attribute to the new `product_brand` [taxonomy](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-brands.php#L237). Then, assign your products to the right brand, remove the legacy attribute and re-save them.

> Since you already handled 2 of them, why not add MPN to core? Just under EAN?

Good idea -- I don't see any open issues for that in the WooCommerce core repository, so I recommend opening one with the above information so we can better track demand.

**Reply by rudtek** [reply]
Will this new brand be a part of the REST API? That could solve my issue. I created my own custom brand (product_brands) and added it to the REST API.  I can see brands attached  when I use a GET for products, and can add exiting terms to a product on a PUT to update a product, but can't seem to add terms of the taxonomy when I POST a NEW product.   Is there a place I could ask for help with a question like this?  I've posted on the woo slack channel, on stackoverflow and haven't received any feedback.  Would a new question on this GIT repository be appropriate?

**Reply by jimjasson** [reply]
@rudtek The new Brands feature will be compatible with the REST API. You may view some examples in the WooCommerce Brands documentation: https://woocommerce.com/document/woocommerce-brands/#rest-api-support -- this document will be moved to core documentation after the release of WooCommerce 9.6. 

If you still have some questions after studying the above examples, then please share a link to your message in the WooCommerce Community Slack and someone from our team will have a look 👀

**Reply by rudtek** [reply]
Yes, I did review that documentation.  There's nothing showing adding an existing brand term to a new product as created through REST.  I'll post again to slack.  Thank you.

**Reply by jimjasson** [reply]
Thanks for the additional details @rudtek! 

The current REST API integration allows you to create, update and read brands in your store, but there isn't an integration with the `products` endpoint that allows you to create a new product with brands or update a product and add new brands. 

This feature didn't exist in the WooCommerce Brands plugin, and we didn't add any new functionality when merging this plugin into core. 

I went ahead and opened a new issue about this: https://github.com/woocommerce/woocommerce/issues/52504 in case you'd like to add some more details about your use case there.

**Reply by qtjohn** [reply]
> > Since you already handled 2 of them, why not add MPN to core? Just under EAN?
> 
> Good idea -- I don't see any open issues for that in the WooCommerce core repository, so I recommend opening one with the above information so we can better track demand.

https://github.com/woocommerce/woocommerce/issues/52332
Thanks!

**Comment by Gregoire-Lesmainsdansleguidon** [comment]
Excellent initiative, i was using attribute, but had issue with google not having the brand in the product sync !

i just have one question, regarding the migration from "brand" as an attribute to the new woocommerce brand
especially on the archive page showing all product of a specific brand, 
the slug of the new "brand page" seems to be /brand/slug-of-the-brand
it seems to conflict with the attribute page /brand/slug-of-brand-as-attribute, as the attribute name is "brands".

did i miss something ?

Grégoire.

**Reply by jimjasson** [reply]
Hey Grégoire, 

Indeed, if you create a custom attribute with 'brand' as the slug and you create a term called 'test', then the URL for this term will be `brand/test`, which is identical to the URL of a brand created under Products > Brands with the name 'test'. 

Similarly, if you create a custom attribute with 'product-category' as the slug, you will notice that the URLs of these attribute terms will be the same as categories defined under Products > Categories. 

WooCommerce doesn't prevent you from using attribute slugs that are identical to slugs reserved by core taxonomies. However, to avoid having pages with the same URLs, I recommend using core's Categories and Brands features, rather than relying on custom attributes. In your case, after porting all your brands to Products > Brands and assigning the new brands to your products, you could delete your custom attribute. Till then, you could disable the Brands module by using this snippet: 

```
add_action( 'init', function() {
    update_option( 'wc_feature_woocommerce_brands_enabled', 'no' );
} );
```

**Reply by Gregoire-Lesmainsdansleguidon** [reply]
Thank you for this very clear answer !
grégoire.
